### PR TITLE
tools/api: replace client to pd http client

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: ref #7300

### What is changed and how does it work?

```commit-message
change tools/api client to HTTP SDK
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)

### Release note

```release-note
None.
```

image

source in log is `tools-api-bench`
```
[2023/12/19 11:27:42.366 +08:00] [INFO] [client.go:315] ["[pd] http client members info changed"] [source=tools-api-bench] [old-member-num=1] [new-member-num=1] [old-addrs="[http://127.0.0.1:2379]"] [new-addrs="[http://127.0.0.1:2379]"] [old-leader-addr-idx=-1] [new-leader-addr-idx=0] [old-leader-addr=] [new-leader-addr=http://127.0.0.1:2379]
```

### CODE REVIEW

File: tools/pd-api-bench/cert_opt.sh
Comment (JmPotato) [inline tools/pd-api-bench/cert_opt.sh:1]: Seems unnecessary change?

Comment (HuSharp) [reply tools/pd-api-bench/cert_opt.sh:1]: I think we need exeute authority

File: tools/pd-api-bench/cases/cases.go [hunk 1]
```diff
@@ -19,21 +19,13 @@ import (
...
 )
 
-var (
-	// PDAddress is the address of PD server.
-	PDAddress string
-	// Debug is the flag to print the output of api response for debug.
-	Debug bool
-)
+// Debug is the flag to print the output of api response for debug.
+var Debug bool
```
Comment (JmPotato) [inline tools/pd-api-bench/cases/cases.go:28]: Maybe merge with the `var` block below.

**Review by JmPotato — APPROVED** [review]

**Comment by HuSharp** [comment]
@CabinfeverB PTAL, thx!

**Review by CabinfeverB — APPROVED** [review]
